dev/scripts/loc.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# Usage: `./loc.sh | sort -n` to get a sorted list of all changed lines.
+
+for user in "Adrian Richter" "sangmeister" "Bastian Rihm" "Elblinator" "Emanuel Schütze" "reiter" "luisa" "Magnus Schieder" "Norman Jäckel" "Oskar Hahn" "Ralf Peschke" "Tobias Hößl" "rrenkert"; do
+	submodules=$(git submodule foreach --quiet git log --since="01-01-2023" --numstat --pretty="%H" --author="$user" | awk 'NF==3 {plus+=$1; minus+=$2} END {printf("%d %d\n", plus, minus)}')
+	main=$(git log --since="01-01-2023" --numstat --pretty="%H" --author="$user" | awk 'NF==3 {plus+=$1; minus+=$2} END {printf("%d %d\n", plus, minus)}')
+	plus=$(( $(echo $submodules | cut -d " " -f 1) + $(echo $main | cut -d " " -f 1) ))
+	minus=$(( $(echo $submodules | cut -d " " -f 2) + $(echo $main | cut -d " " -f 2) ))
+	total=$((plus + minus))
+	echo "$total $user: +$plus, -$minus"
+done
```

Comment:
```suggestion
# Usage: `./loc.sh [<start_date> (default: 01-01-%Y)]`

# The list of authors whose names must be merged.
# "<main name>:<name to be merged>"
merge_list=(
    "Joshua Sangmeister:jsangmeister"
    "luisa-beerboom:Luisa"
    "reiterl:Ludwig Reiter"
)

merge_authors() {
    awk -v p1="$1" -v p2="$2" -F '|' '
        $2 ~ p1 {total+=$1; plus+=$3; minus+=$4; next}
        $2 ~ p2 {total+=$1; plus+=$3; minus+=$4; next}
        {print $0}
    END {printf(" %d | %s | %d |%d\n", total, p1, plus, minus) }' <<< "$list"
}

start_date=$(date +'01-01-%Y')
[[ -z $1 ]] || start_date=$1

list=$(while read -r author; do
    submodules=$(git submodule foreach --quiet git log --since="$start_date" \
        --numstat --pretty="%H" --author="$author" |\
        awk 'NF==3 {plus+=$1; minus+=$2} END {printf("%d %d\n", plus, minus)}')
    main=$(git log --since="$start_date" --numstat --pretty="%H" --author="$author" |\
        awk 'NF==3 {plus+=$1; minus+=$2} END {printf("%d %d\n", plus, minus)}')
    plus=$(( $(cut -d " " -f 1 <<< $submodules) + $(cut -d " " -f 1 <<< $main) ))
    minus=$(( $(cut -d " " -f 2 <<< $submodules) + $(cut -d " " -f 2 <<< $main) ))
    total=$((plus + minus))
    echo " $total | $author | $plus | $minus"
done <<< $(git submodule foreach --quiet git log --format='%aN' --since="$start_date" \
    | sort -u | grep -v "\[bot\]$"))

# For authors who have several names...
for m in "${merge_list[@]}"; do
    list=$(merge_authors "$(cut -d ":" -f 1  <<< $m)" "$(cut -d ":" -f 2 <<< $m )")
done

echo "All changed lines per author since $start_date:"
printf "$list\n" | sort -nr | column --table-columns " TOTAL, AUTOR, PLUS, MINUS" \
    --table-order " AUTOR, TOTAL, PLUS, MINUS" -ts '|' -o '|'
```

Reply:
Looks good! Some additional changes were necessary or were liked better by me:
- fixed a bug where some commits would be counted twice (e.g. "Joshua Sangmeister" and "jsangmeister" would both match a commit authored by me, since the commit author would be "Joshua Sangmeister <33004050+jsangmeister@users.noreply.github.com>") by prepending the author filter with a `^`
- extended merge list for @bastianjoel and @peb-adr
- abused merge list to show real names for all authors
- fixed typo and table spacing
- included bots to show that they are only human, too